langchain4j-coherence/src/main/java/dev/langchain4j/store/embedding/coherence/CoherenceEmbeddingStore.java

```diff
@@ -0,0 +1,526 @@
+        addInternal(id, embedding, segment);
+        return id.toString();
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        return addAll(embeddings, null);
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<DocumentChunk.Id> keys = generateKeys(embeddings.size());
+        addAllInternal(keys, embeddings, embedded);
+        return keys.stream().map(DocumentChunk.Id::toString).collect(Collectors.toList());
+    }
+
+    @Override
+    public void remove(String id) {
+        if (id == null || id.isBlank()) {
+            throw new IllegalArgumentException("id cannot be null or blank");
+        }
+
+        documentChunks.remove(DocumentChunk.Id.parse(id));
+    }
+
+    @Override
+    public void removeAll(Collection<String> ids) {
+        if (ids == null || ids.isEmpty()) {
+            throw new IllegalArgumentException("ids cannot be null or empty");
+        }
+
+        documentChunks.keySet().removeAll(ids.stream().map(DocumentChunk.Id::parse).collect(Collectors.toSet()));
+    }
+
+    @Override
+    public void removeAll(dev.langchain4j.store.embedding.filter.Filter filter) {
+        if (filter == null) {
+            throw new IllegalArgumentException("filter cannot be null");
+        }
+        
+        Filter<DocumentChunk> chunkFilter = CoherenceMetadataFilterMapper.map(filter);
+        documentChunks.invokeAll(chunkFilter, CacheProcessors.removeBlind());
+    }
+
+    @Override
+    public void removeAll() {
+        documentChunks.truncate();
+    }
+
+    @Override
+    public EmbeddingSearchResult<TextSegment> search(EmbeddingSearchRequest request) {
+        Embedding queryEmbedding = request.queryEmbedding();
+        if (forceNormalize) {
+            queryEmbedding.normalize();
+        }
+        double minScore = request.minScore();
+        boolean checkMin = minScore != 0.0f;
+        Filter<DocumentChunk> filter = CoherenceMetadataFilterMapper.map(request.filter());
+        Float32Vector vector = new Float32Vector(queryEmbedding.vector());
+
+        SimilaritySearch<DocumentChunk.Id, DocumentChunk, float[]> aggregator = new SimilaritySearch<>(EXTRACTOR, vector, request.maxResults());
+        if (filter != null) {
+            aggregator = aggregator.filter(filter);
+        }
+            
+        List<QueryResult<DocumentChunk.Id, DocumentChunk>> results = documentChunks.aggregate(aggregator);
+        List<EmbeddingMatch<TextSegment>> matches = new ArrayList<>();
+
+        for (QueryResult<DocumentChunk.Id, DocumentChunk> result : results) {
+            double score = RelevanceScore.fromCosineSimilarity(1.0f - result.getDistance());
+            if (checkMin && score < minScore) {
+                continue;
+            }
+
+            DocumentChunk.Id id = result.getKey();
+            DocumentChunk chunk = result.getValue();
+
+            boolean fHighlyRelevant = score > 0.75f;
```

Comment:
where does this value comes from? Why is treatment different in this case?

Reply:
Tbh, I have no idea... 

@thegridman Thoughts?

Reply:
Ah, actually, I do -- this is something I was playing with in our RAG implementation, but I don't think it's necessary (or desired) here. I'll remove it, good catch.